### ISSUE TITLE
`del` statement in a loop causes false errors

### ISSUE THREAD

**Issue by Molkree** [microsoft/pyright#4346]
**Describe the bug**
I'm getting "Argument type is partially unknown" errors when I use `del` statement in a loop on a list element.

**To Reproduce**
See the code below.

**Expected behavior**
No errors reported.

**Screenshots or Code**
```python
lst: list[tuple[int, int]] = [(0, 0)]
for _ in range(1):
    lst[-1] = lst[-1][1], lst[-1][0]  # error here
    del lst[-1]  # this causes the error
    # lst.pop()  # no error if I use `pop` instead of `del`

```

![image](https://user-images.githubusercontent.com/12860284/208212692-05ead6d0-ecf9-4ac3-97ce-adcbdb21ca60.png)

```
Argument type is partially unknown
  Argument corresponds to parameter "__o" in function "__setitem__"
  Argument type is "tuple[int | Unknown, int | Unknown]"
```

**VS Code extension or command-line**
Pylance v2022.12.21

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.285, which I just published. It will also be included in a future release of pylance.
